.github/workflows/build-test-osx-arm64.yaml

```diff
@@ -32,7 +32,7 @@ jobs:
         "ghcr.io/cirruslabs/macos-monterey-xcode:latest",
       ]
     env:
-      OPAM_SWITCH_NAME: "4.14.0"
+      OPAM_SWITCH_NAME: "5.0.0"
```

Comment:
Should we switch to 5.1.0 directly (which was released a few weeks ago)?

Reply:
I might prefer that we hold that one off for later -- this would require me to basically revisit every change I've made, and another PR to `ocaml-layer` to boot (which might induce more dependency hell).

Let's get this milestone in quickly, while it's easy to do, and worry about 5.1 soon after.

Reply:
ok